<launch>
    <arg name="namespace" default="$(env UAV_NAMESPACE)"/>
    <arg name="input_trajectory" default="topp/input/trajectory"/>

    <arg name="output_point" default="position_hold/trajectory"/>
    <arg name="odom" default="mavros/global_position/local"/>
    <arg name="trajectory_flag" default="topp/trajectory_flag"/>

    <!-- Trajectory generation -->
    <node name="trajectory_gen" pkg="topp_ros" type="generate_toppra_trajectory.py" 
        output="screen" ns="$(arg namespace)"/>

    <!-- Trajectory point publisher -->
    <node name="trajectory_publisher" pkg="uav_ros_control" type="generate_from_traj_input.py" 
        output="screen" ns="$(arg namespace)">
        <remap from="input/trajectory" to="$(arg input_trajectory)"/>
        <remap from="output/point" to="$(arg output_point)"/>
        <remap from="odometry" to="$(arg odom)"/>
        <remap from="trajectory_flag" to="$(arg trajectory_flag)"/>
    </node>

</launch>